comp/logs/agent/config/integration_config.go

```diff
@@ -199,6 +218,84 @@ func (c *LogsConfig) PublicJSON() ([]byte, error) {
 	})
 }
 
+// UnmarshalYAML is a custom unmarshalling function is needed for string array fields to split comma-separated values.
+func (c *LogsConfig) UnmarshalYAML(node *yaml.Node) error {
+	yamlToFieldCache := buildYamlToFieldCache(c)
+	for i := 0; i < len(node.Content); i += 2 {
+		key := node.Content[i].Value
+		valueNode := node.Content[i+1]
+		structField, exists := yamlToFieldCache[key]
+		if !exists {
+			return fmt.Errorf("field %s not found in struct", key)
+		}
+
+		if structField.CanSet() {
+			switch valueNode.Kind {
+			// A ScalarNode represents a single value (string, int, *bool, bool, float64, or slice
+			// represented as a string).
+			// A SequenceNode a multiple(list) of such values.
+			case yaml.ScalarNode:
+				if structField.Kind() == reflect.String {
+					structField.SetString(valueNode.Value)
+				} else if structField.Kind() == reflect.Int {
+					intValue, err := strconv.Atoi(valueNode.Value)
+					if err == nil {
+						structField.SetInt(int64(intValue))
+					}
+				} else if structField.Kind() == reflect.Slice {
+					//an example of a slice when the input is a scalar node:
+					//tags: a, b:c -> []string{"a", "b:c"}
+					rawValue := strings.Split(valueNode.Value, ",")
+					strSlice := make([]string, 0, len(rawValue))
+					for _, v := range rawValue {
+						strSlice = append(strSlice, strings.TrimSpace(v))
+					}
```

Comment:
You could replace in the original slice instead of creating a new one:

```suggestion
					slice := strings.Split(valueNode.Value, ",")
					for idx, v := range slice {
						slice[idx] = strings.TrimSpace(v)
					}
```